src/logplex_tlssyslog_drain.erl

```diff
@@ -84,20 +87,44 @@
 %% API Function Definitions
 %% ------------------------------------------------------------------
 
-start_link(ChannelID, DrainID, DrainTok,
-           #ex_uri{scheme="syslog+tls",
-                   authority=#ex_uri_authority{host=Host, port=Port}}) ->
-    start_link(ChannelID, DrainID, DrainTok, Host, Port).
-
-start_link(ChannelID, DrainID, DrainTok, Host, Port) ->
+start_link(ChannelID, DrainID, DrainTok, Uri) ->
+    {Host, Port, Insecure} = unpack_uri(Uri),
     gen_fsm:start_link(?MODULE,
                        [#state{drain_id=DrainID,
                                drain_tok=DrainTok,
                                channel_id=ChannelID,
+                               uri=Uri,
                                host=Host,
-                               port=Port}],
+                               port=Port,
+                               insecure=Insecure}],
                        []).
 
+unpack_uri(#ex_uri{scheme="syslog+tls",
+                  authority=#ex_uri_authority{host=Host, port=Port},
+                  fragment=Fragment}) ->
+    Opts =  parse_fragment(Fragment),
+    Insecure = case proplists:lookup("insecure", Opts) of
+                   {"insecure", "true"} -> true;
+                   _ -> false
+               end,
+    {Host, Port, Insecure}.
+
+parse_fragment(undefined) ->
```

Comment:
This seems like a lot of work to parse the fragment - there's no particular standard for the encoding in the fragment, so you could just match `Fragment =:= "#insecure"` or `Fragment =:= "insecure=true"` here right? (If the 'insecure' thing isn't set in stone, you could also use something like 'skip_cert_validation' which would require less devcenter documentation to explain the purpose of this option).